src/SonarScanner.MSBuild.PreProcessor/WebClientDownloader.cs

```diff
@@ -32,29 +32,35 @@ namespace SonarScanner.MSBuild.PreProcessor
 {
     public class WebClientDownloader : IDownloader
     {
+        // This is a temporary solution untill we upgrade to .net framework 4.8.
+        private const SecurityProtocolType Tls13 = (SecurityProtocolType)12288;
+        private const SecurityProtocolType SystemDefault = 0;
+
         private readonly ILogger logger;
         private readonly HttpClient client;
 
         public WebClientDownloader(string userName, string password, ILogger logger, string clientCertPath = null, string clientCertPassword = null)
         {
-            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
-            if (password == null)
+            if (ServicePointManager.SecurityProtocol != SystemDefault)
```

Comment:
Could you please clarify a bit? I am not sure I understand what you are suggesting here.

Reply:
Can we tell if the runtime is using a vulnerable TLS version?
It would be nice to log a warning.
Do we have a ticket to log a warning if the runtime has a vulnerable TLS version?

Reply:
I added some logging as discussed offline.